src/windows.rs

```diff
@@ -0,0 +1,464 @@
+        DISPLAY_BRIGHTNESS, IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS,
+        IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS, IOCTL_VIDEO_SET_DISPLAY_BRIGHTNESS,
+    },
+    Foundation::{CloseHandle, BOOL, HANDLE, LPARAM, PWSTR, RECT},
+    Graphics::Gdi::{
+        EnumDisplayDevicesW, EnumDisplayMonitors, GetMonitorInfoW, DISPLAY_DEVICEW, HDC, HMONITOR,
+        MONITORINFO, MONITORINFOEXW,
+    },
+    Storage::FileSystem::{
+        CreateFileW, FILE_ACCESS_FLAGS, FILE_FLAGS_AND_ATTRIBUTES, FILE_SHARE_MODE, OPEN_EXISTING,
+    },
+    System::Diagnostics::Debug::{GetLastError, ERROR_ACCESS_DENIED, ERROR_NOT_FOUND, WIN32_ERROR},
+    System::SystemServices::{DeviceIoControl, GENERIC_READ, GENERIC_WRITE},
+    UI::WindowsAndMessaging::EDD_GET_DEVICE_INTERFACE_NAME,
+};
+
+#[allow(non_snake_case)]
+#[derive(Debug)]
+pub struct Brightness {
+    hPhysicalMonitor: HANDLE,
+    file_handle: HANDLE,
+    device_name: String,
+    device_description: String,
+    device_id: String,
+    device_key: String,
+    device_string: String,
+}
+
+impl Drop for Brightness {
+    fn drop(&mut self) {
+        unsafe {
+            DestroyPhysicalMonitor(self.hPhysicalMonitor);
+            CloseHandle(self.file_handle);
+        }
+    }
+}
+
+#[async_trait]
+impl crate::Brightness for Brightness {
+    async fn device_name(&self) -> Result<String, Error> {
+        Ok(self.device_name.clone())
+    }
+
+    async fn device_info(&self) -> Result<HashMap<String, String>, Error> {
+        Ok(hashmap! {
+            "device_description".to_string() => self.device_description.clone(),
+            "device_id".to_string() => self.device_id.clone(),
+            "device_key".to_string() => self.device_key.clone(),
+            "device_string".to_string() => self.device_string.clone(),
+        })
+    }
+
+    async fn get(&self) -> Result<u32, Error> {
+        let ioctl_query = ioctl_query_supported_brightness(self);
+        return match ioctl_query {
+            Ok(_) => ioctl_query_display_brightness(self).map_err(|e| e.into()),
+            Err(e) if e == ERROR_NOT_FOUND || e == WIN32_ERROR(6) => {
+                ddcci_get_monitor_brightness(self)
+                    .map(|b| b.get_current_percentage())
+                    .map_err(|e| e.into())
+            }
+            Err(e) => Err(SysError::GettingBrightnessFailed {
+                device_name: self.device_name.clone(),
+                function_name: "IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS",
+                last_error: e,
+            }
+            .into()),
+        };
+    }
+
+    async fn set(&mut self, percentage: u32) -> Result<(), Error> {
+        let ioctl_query = ioctl_query_supported_brightness(self);
+        match ioctl_query {
+            Ok(levels) => {
+                let new_value = levels.get_nearest(percentage);
+                ioctl_set_display_brightness(self, new_value).map_err(|e| e.into())
+            }
+            Err(e) if e == ERROR_NOT_FOUND => {
```

Comment:
Does `Err(ERROR_NOT_FOUND)` compile?

Reply:
Yep everything compiles, you could check and enforce this with a github action (free for open source)
You can see it compiling as a dependency here https://github.com/jacob-pro/solar-screen-brightness/runs/3464706708?check_suite_focus=true#step:6:410

Reply:
Do you mean you've tried the suggestion I recommended (`Err(ERROR_NOT_FOUND)` instead of `Err(e) if e == ERROR_NOT_FOUND`) and the suggestion compiles?

Reply:
Oh sorry I misread, will check

Reply:
Based on the documentation, I expect my suggestion to compile, which avoids the need for a match guard.

Reply:
> Oh sorry I misread, will check

No problem. Thank you for checking.

Reply:
yep, works great

Reply:
Thank you.